### ISSUE TITLE
shelljs should not kill process if node call throws exception

### ISSUE THREAD

**Issue by TimBarham** [shelljs/shelljs#473]
### Node version:
4.4.7

### ShellJS version (the most recent version/Github branch you see the bug on):
0.7.0

### Operating system:
Windows 10

### Description of the bug:
I'm working on code that calls shelljs mkdir to create a directory. On Windows, the call to fs.mkdir() sometimes fails because of permission issues or whatever. This is a failure that I should be able to catch with a try/catch. But shelljs catches the error, outputs it to the console and exits the process ([here](https://github.com/shelljs/shelljs/blob/master/src/common.js#L333)). A utility like shelljs should never take such drastic action - it should be up to me the consumer to decide what to do with such an error.

### Example ShellJS command to reproduce the error:
On windows:

```javascript
var shelljs = require('shelljs');
shelljs.mkdir('c:\\Program Files\\foo');
```

It should just throw in this situation so I can catch the error and decide what to do.

**Comment by ariporad** [comment]
Hmm... Good point.

@nfischer: I think this behavior should change. Thoughts?

**Comment by nfischer** [comment]
This should report an error that can be checked with `shell.error()`, to stay consistent with the rest of the API I think

**Comment by nfischer** [comment]
@TimBarham fixed in #474. This doesn't test on Windows, since `shell.chmod()` is limited in functionality on Windows. The fix should work on all systems, however.

**Comment by TimBarham** [comment]
Thanks @nfischer. Appreciate the fix.

**Comment by nfischer** [comment]
@TimBarham This should be available in the latest ShellJS version, v0.7.2.

**Comment by maikokuppe** [comment]
There is a case where this fix does not apply:

    var shelljs = require('shelljs');
    shelljs.touch('someFile');
    shelljs.mkdir('someFile/someDir');

This results in the error

    ShellJS: internal error
    Error: ENOTDIR: not a directory, mkdir 'someFile/someDir'

which cannot be caught by try ... catch

**Comment by nfischer** [comment]
@maikokuppe the fix hasn't been released yet. It should cover all such cases, however.

To clarify, `ShellJS: internal error` indicates you hit a bug in ShellJS. If you do some operation and get that error, please file a separate bug report describing what you did so we can fix that scenario and provide sensible behavior. That `mkdir()` case would be a great example of something you should file a bug report for (and please feel free to do so).

The change which closed this particular bug is to just stop killing the process, and allow you to catch the exception (will be released in ShellJS v0.8).
